### PR TITLE
Closes #846. Add detekt rule to check license header in kotlin files.

### LINKED ISSUE #846: Add an automated check for license headers on top on new files (lint rule, detekt check, ..)
We should have an automated check for that verifies that  new files have the license headers on top -  if not the PRs fail automatically.

### PR DESCRIPTION
**Pull request by dector**
---

  - Created new component for custom detekt plugins (rules, reporters etc).
  - Created custom `AbsentOrWrongFileLicense` rule. Parsing file header because detekt don't recognize `\* ... *\` atop as a comment.
  - Created custom `mozilla-rules` ruleset for detekt.
  - Activated the rule.
  - Generated new baseline file (btw, there are some false-positives because we are checking exact license string and in few files leading spaces are missed). Follow-up issue: #4460.

### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/tooling/detekt/build.gradle [hunk 1]
```diff
@@ -0,0 +1,28 @@
+apply plugin: 'java-library'
```
Comment (pocmo) [inline components/tooling/detekt/build.gradle]: nit: license header 😂 😝

File: build.gradle [hunk 2]
```diff
@@ -19,7 +19,7 @@ buildscript {
 }
 
 plugins {
-    id("io.gitlab.arturbosch.detekt").version("1.0.1")
+    id("io.gitlab.arturbosch.detekt").version("1.0.1")  // Check version number in `Versions.detekt`
```
[the remark this answers is not in the document]
Comment (dector) [reply build.gradle]: It's reference to
```
object Versions {
    val detekt = "..."
}
```.

File is (`Dependencies.kt`).

File: components/tooling/detekt/build.gradle (hunk 1 above)
Comment (dector) [reply components/tooling/detekt/build.gradle]: :ok_hand: :smile_cat:

**Comment by dector** [comment]
@pocmo I also thought about moving this rule to `style` section. But, probably for visibility reasons it's better to keep ruleset separately (as it's only for internal use).

**Comment by dector** [comment]
@pocmo rebased. :)

**Pull request merged**